### PR TITLE
Timeline configuration, timeline rework + support for projectdeliveries

### PR DESCRIPTION
**Pull request by Remi749**
### Your checklist for this pull request

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [X] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [X] Check the commit's or even all commits' message 
- [X] Check if your code additions will fail linting checks
- [X] Remember: Add issue description to [CHANGELOG](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the **ID of the issue** associated with this PR
- [X] Documentation: Have a look at the [PP365 User manual](https://puzzlepart.github.io/prosjektportalen-manual/) and consider the need for updates to be made. Updates can be done directly into the 'Kladd' branch or by providing information to test team for implementation.

## Additions

- Added button to 'Konfigurasjon av Prosjektportalen' page.
- Added page with list for use when configuring the timeline webpart

- Added the following to Upgrade script:
  - Migrate content from 'TimelineType' to 'GtTimelineTypeLookup'
  - Migrate content from  'SiteIdLookup' to 'GtSiteIdLookup'

### Provision template additions:

Added button (QuickLink) for 'Tidslinjekonfigurasjon' list to 'Konfigurasjon av Prosjektportalen' page.

**New SiteFields**:
*DisplayNames, Descriptions added to resource files*
- GtHexColor
- GtElementType
- GtShowElementPortfolio
- GtShowElementProgram
- GtTimelineFilter
- GtSiteIdLookup
- GtTimelineTypeLookup

**New ContentTypes**:
*Name, Descriptions added to resource files*
- Tidslinjeelement
- Tidslinjekonfigurasjonselement

**New list**:
*Title, Url added to resource files*
- Tidslinjekonfigurasjon
image

**Updates to Tidslinjeinnhold list**
- Added local Fields to resources
- New contenttype and SiteFields
- Added mapping for old to new structure in upgrade script (TODO)

### UI/UX fixes:
- Adjusted beak and top-border color of the selected item's callout dialog (see 'after images' underneath)
- Adjust arrow from callout to actually hit the selected item on the timeline

| Before | After |
| ------- | ----- |
| image | image |

- Fit the callout dialogs within the actual bounds of the timeline
  - This fixes the issue with the dialog appearing far left when having a long element or adjusted the timeline so the element is far on the left side (see images)

| Before | After |
| ------- | ----- |
| image | image |

### Projectdeliveries items on timeline (Project level only, for now)

The timeline has been expanded so that the projectdeliveries list content can be displayed and connected to a project on the timeline.

ProjectTimeline webpart properties:
- Enable/disable project deliveries for timeline (default: disabled)
- Defined listname for project deliveries (if customer has custom project deliveries list)
- Define the title used for the type in timeline configuration list
image

Timeline element and Details callout

| With custom configuration | Default |
| ------- | ----- |
| image | image |

See issues for more information

#### After upgrade

1. Check that the data has been correctly migrated in the 'Tidslinjeinnhold' list

#### Timeline configuration

1. Try to change values in the configuration list and see that the changes occur on:
 a. ProjectTimeline (Portfolio level)
 b. ProjectTimeline (Project level)
 c. ProgramTimeline
3. Introduce new elements in the Timline configuration list and see what happens when you change element types, etc...

#### Projectdeliveries

1. Activate projectdeliveries in the webparts props
2. Define listname and element title or leave default if there is not a custom list for this
3. Create an element in the timeline configuration list for projectdeliveries with the same title as defined in the webpart props

### Update of documentation
- [x] Update user manual if needed

### Relevant issues (if applicable)

#678 #679

### CODE REVIEW

File: SharePointFramework/PortfolioWebParts/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pp365-portfoliowebparts",
-  "version": "1.5.4",
+  "version": "1.5.4-749.2",
```
Comment (olemp) [inline SharePointFramework/PortfolioWebParts/package.json:3]: @Remi749 What is 749? Issue key? PR key?

Comment (olemp) [reply SharePointFramework/PortfolioWebParts/package.json:3]: PR key preferably @Remi749

Comment (olemp) [reply SharePointFramework/PortfolioWebParts/package.json:3]: But it's not super important. It just gives some context.

Comment (Remi749) [reply SharePointFramework/PortfolioWebParts/package.json:3]: Ait, I'll keep that in mind 👍

**Review by olemp — APPROVED** [review]

**Comment by Remi749** [comment]
Thanks for merging! I see that the old "Type" and "Prosjektilknytning" fields are in the "New" dialog when creating a new element. I'm going to look into this.

**Pull request merged**